reana_server/config.py

```diff
@@ -41,6 +41,8 @@
 REANA_COMPLEXITY_JOBS_MEMORY_LIMIT = os.getenv("REANA_KUBERNETES_JOBS_MEMORY_LIMIT")
 """Maximum memory limit for user job containers for workflow complexity estimation."""
 
+REANA_WORKFLOW_SCHEDULING_POLICY = os.getenv("REANA_WORKFLOW_SCHEDULING_POLICY")
```

Comment:
What about falling back explicitly to "fifo" in case the variable does not exist for a reason or another?

```python
REANA_WORKFLOW_SCHEDULING_POLICY = os.getenv("REANA_WORKFLOW_SCHEDULING_POLICY", "fifo")
```
        
          
            
            
            

    

  
      
        
        
        

          
          
          
          
          
          
          



        
  
  
      
        
```

Reply:
A couple of things:
- If the [default value is `balanced`](https://github.com/reanahub/reana/pull/511/files#diff-8a781e33aeeaa4deddd5a0f1fd47e3c6e85de8f7e65e34916f12eef80fa447efR74) why should we fall back to `fifo`?
- Perhaps we can perform the validation already here and fall back to a known value to prevent [raising an exception](https://github.com/reanahub/reana-server/pull/377/files#diff-dd451815d7362d3092b6e4d2bc6880a2fe0698d82d2415e90b8adc23181eb3faR52)? OT1H it's good to be aware that a wrong policy value was passed but OTOH it'd be nice to prevent the cluster from malfunctioning if a wrong config is set.

Reply:
Yeah, the default should be the same, "fifo" or "balanced" everywhere.  I guess we could stay on the conservative "fifo" value until we have full CWL DAG complexity set up... and use "balanced" only at CERN.

Reply:
>     * Perhaps we can perform the validation already here and fall back to a known value to prevent [raising an exception](https://github.com/reanahub/reana-server/pull/377/files#diff-dd451815d7362d3092b6e4d2bc6880a2fe0698d82d2415e90b8adc23181eb3faR52)? OT1H it's good to be aware that a wrong policy value was passed but OTOH it'd be nice to prevent the cluster from malfunctioning if a wrong config is set.

I think if we fallback silently it could be tricky to spot it in case some user makes a typo while defining the policy. Imagine someone who set "blanced" would run "fifo" without knowing it